syntax = "proto3";

package local_server;

service LocalServer {
  rpc Ping(EmptyMessage) returns (EmptyMessage) {}
  rpc SendTextMessage(TextMessage) returns (EmptyMessage) {}
  rpc UpdatePos(Pos) returns (stream Pos) {}
  rpc FetchJob(RobotID) returns (Job) {}
}

message EmptyMessage {

}

message TextMessage {
    string text = 1;
}

message Point {
    int32 x = 1;
    int32 y = 2;
}

message Pos {
    string name = 1;
    Point point = 2;
}

message RobotID {
    string name = 1;
}

message Job {
    string job_type = 1;
    string charger = 2;
    string source_station = 3;
    string target_station = 4;
}
